Mathlib/LinearAlgebra/Trace.lean

```diff
@@ -10,6 +10,7 @@ import Mathlib.LinearAlgebra.TensorProductBasis
 import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
 import Mathlib.LinearAlgebra.FreeModule.Finite.Rank
 import Mathlib.LinearAlgebra.Projection
+import Mathlib.LinearAlgebra.FreeModule.PID
```

Comment:
I'm a bit skeptical about this extra import: it feels like we're bringing in quite a bit of theory that would not be needed to simply define the trace. What do you think of putting your additions in a new file?

Reply:
I had the same concern but I couldn't quite convince myself that such a file had a future beyond this little lemma.

However, I'm really on the fence: if you think a new file is better, I'll happily take that approach.

Reply:
I agree the tradeoff is not obviously on either the splitter or the merger side. In this case, splitting files is going to be cheaper to undo (move the imports and declarations into one file) compared to merging (where you'd have to determine per declaration which file would be most appropriate), so I'd advocate splitting when in doubt.

This is actually something I learned from Linus Torvalds: if a choice is unclear, go for the one that is easiest to revert :)

Reply:
OK I've split it out!